crates/downloader/src/lib.rs

```diff
@@ -110,16 +99,21 @@ fn get_blobs_from_beacon_response(
 }
 
 pub struct Downloader {
-    /// Ethereum provider client.
-    provider: RootProvider<Http<reqwest::Client>>,
     /// HTTP client for downloading blobs.
     http: reqwest::Client,
     /// API URL for the beacon.
     beacon_url: Url,
     /// Contract to monitor for new blobs.
     target_contract: Address,
-    /// Current block number.
-    current_block: u64,
+    /// Stream of blocks downloaded from the L1.
+    download_stream:
+        Pin<Box<dyn Stream<Item = anyhow::Result<Block>> + Send + Sync + 'static>>,
```

Comment:
I'm curious if we _have_ to Pin<Box<dyn T>> the stream, what if we made the `Downloader` generic over something that implements `Stream` instead?

Reply:
looks like we have to pin it, feel free to ignore